pkg/cmd/roachtest/cluster.go

```diff
@@ -3010,6 +3011,27 @@ func archForTest(ctx context.Context, l *logger.Logger, testSpec registry.TestSp
 	return arch
 }
 
+// bucketCachedClusterVMs buckets cachedCluster.VMs by provider.
+func bucketCachedClusterVMs(cachedCluster *cloud.Cluster) map[string][]vm.VM {
+	providerToVMs := make(map[string][]vm.VM)
+	for _, vm := range cachedCluster.VMs {
+		providerToVMs[vm.Provider] = append(providerToVMs[vm.Provider], vm)
+	}
+	return providerToVMs
+}
+
+// getAvailableClusters returns a slice of the names of all available clusters,
+// or an error if the specified cluster cannot be found.
+func getAvailableClusters(c *clusterImpl) ([]string, error) {
```

Comment:
>// getAvailableClusters returns a slice of the names of all available clusters,
// or an error if the specified cluster cannot be found.

This seems off, if we're returning a list of all available clusters, then finding a specific cluster sounds like it doesn't have anything to do with that.

I think what you actually want here is this to be `getCachedCluster` which checks if `c.name` is is in `roachprod.CachedCluster()` and returns an error if not found with the list of available clusters.

Reply:
Nice catch! Updated